examples/js/vr/VRController.js

```diff
@@ -0,0 +1,756 @@
+
+		style: 'vive',
+		buttons: [ 
+
+
+			//  Vive’s thumpad is both a 2D trackpad and a button. We can
+			//  1. touch it -- simply make contact with the trackpad (binary)
+			//  2. press it -- apply force to depress the button (binary)
+			//  3. get XY values for the point of contact on the trackpad.
+			//  X axis: -1 = Left,   +1 = Right
+			//  Y axis: -1 = Bottom, +1 = Top
+
+			'thumbpad', 
+
+
+			//  Vive’s trigger offers a binary touch and a
+			//  gradient of “pressed-ness” values from 0.0 to 1.0.
+			//  Here’s my best guess at the trigger’s internal rules:
+			//  if( value > 0.00 ) touched = true else touched = false
+			//  if( value > 0.51 ) pressed = true   THRESHOLD FOR TURNING ON
+			//  if( value < 0.45 ) pressed = false  THRESHOLD FOR TURNING OFF
+
+			'trigger', 
+
+			
+			//  Each Vive controller has two grip buttons, one on the left and one on the right.
+			//  They are not distinguishable -- pressing either one will register as a press
+			//  with no knowledge of which one was pressed.
+			//  This value is binary, it is either touched/pressed (1) or not (0)
+			//  so no need to track anything other than the pressed boolean.
+
+			'grips', 
+
+
+			//  The menu button is the tiny button above the thumbpad (NOT the one below it).
+			//  It’s simple; just a binary on / off press.
+
+			'menu'
+		],
+		primary: 'trigger'
+	},
+	'Oculus Touch (Right)': {
+
+		style: 'rift',
+		buttons: [
+
+
+			//  Rift’s thumbstick has axes values and is also a button, 
+			//  similar to Vive’s thumbpad.
+			//  But unlike Vive’s thumbpad it only has a binary touch value. 
+			//  The press value is never set to true.
+			//  X axis: -1 = Left, +1 = Right
+			//  Y axis: -1 = Top,  +1 = Bottom  NOTE THIS IS FLIPPED FROM VIVE!
+
+			'thumbstick',
+
+
+			//  Rift’s trigger is twitchier than Vive’s.
+			//  Compare these threshold guesses to Vive’s trigger:
+			//  if( value > 0.1 ) pressed = true   THRESHOLD FOR TURNING ON
+			//  if( value < 0.1 ) pressed = false  THRESHOLD FOR TURNING OFF
+
+			'trigger',
+
+
+			//  Rift’s grip button follows the exact same pattern as the trigger.
+
+			'grip',
+
+
+			//  Rift has two old-school video game buttons, A and B.
+			// (For the left-hand controller these are X and Y.)
+			//  They report separate binary on/off values for both touch and press.
+
+			'A', 'B',
+
+
+			//  Rift has an inert base “button” that’s really just a resting place
+			//  for your thumbs and only reports a binary on/off for touch.
+
+			'thumbrest'
+		],
+		primary: 'trigger'
+	},
+	'Oculus Touch (Left)': {
+
+		style: 'rift',
+		buttons: [
+
+			'thumbstick',
+			'trigger',
+			'grip',
+			'X', 'Y',
+			'thumbrest'
+		],
+		primary: 'trigger'
+	}
+}
+
+
+
+
+    ///////////////////
+   //               //
+  //   Arm Model   //
+ //               //
+///////////////////
+
+
+//  Adapted from Boris’ code in a hurry -- many thanks!
+
+const 
+HEAD_ELBOW_OFFSET       = new THREE.Vector3(  0.155, -0.465, -0.15 ),
```

Comment:
Although it’s a bit verbose instead of wrapping it all in a closure, I made these formerly-global variables into properties of `OrientationArmModel()` because I felt it better expressed their purpose. But what do you think? (That change is in [the latest commit](https://github.com/mrdoob/three.js/pull/10991/commits/a0fab5a02c629fae26262f566df0639af838b136) around line 581.)